packages/mattermost-redux/src/store/helpers.ts

```diff
@@ -15,18 +13,7 @@ export function createReducer(baseState: GlobalState, ...reducers: Reducer[]): R
     reducerRegistry.setReducers(Object.assign({}, ...reducers));
     const baseReducer = combineReducers(reducerRegistry.getReducers());
 
-    // Root reducer wrapper that listens for reset events.
-    // Returns whatever is passed for the data property
-    // as the new state.
-    const offlineReducer = (state: GlobalState, action: Action) => {
-        if ('type' in action && 'data' in action && action.type === General.OFFLINE_STORE_RESET) {
-            return baseReducer(action.data || baseState, action);
```

Comment:
This is the first part of the important change in this PR. On logout, we used to overwrite the whole state tree which seems incorrect when there's some state we might want to save like information about the window size and the client config.

I think the original reason for doing this might have been that there was also some state outside our control which came from redux-offline (which is why this code refers to "offline" a few times). Since we haven't used that in a while, we don't need that part either